### ISSUE TITLE
Suggestion - Deployment folder Settings

### ISSUE THREAD

**Issue by ZeroMaki** [Nexus-Mods/Vortex#6116]
Response to https://github.com/Nexus-Mods/Vortex/issues/6113 since prematurely closed.

That has nothing to do with it, and thats my point exactly, you are NOT and I repeat NOT deploying where game dictates, the game has 2!! mod folders, both packages/core/override, the one you deploy to is documents which are always on C drives unless people move it, if game is D drive, then interdrive loading. You change it to game folder, a change already custom done.

function queryModPath(gamePath) { return path.join(gamePath, 'packages', 'core', 'override'); game path, not documents. Has nothing to do with how mods are packaged and such.

So when advanced mode is active, allow the option to set deployment folder as you can with staging folder, advanced is for the advanced..

**Comment by TanninOne** [comment]
The use case is way too obscure to make it a general-purpose features, out of the >100 games we support now there are maybe 2 which support mods to be installed in different (equivalent) locations.

However: queryModPath is a function, not a fixed value, exactly so custom code can influence the return value.
It should therefore be possible for the dragonage 2 extension to register its own settings flag that toggles the mod folder and then have the queryModPath function return the appropriate directory based on that setting.

I'll leave this open as a feature request but I'll say now: I don't think we will get around to implementing this any time soon, it's just way too obscure a requirement and we have too much other stuff to do so if anyone external wants to look into this, be our guest.

**Comment by ZeroMaki** [comment]
Well could you at least modify the extension to the game path then by default? I'm sure we can all agree it would be cleaner.

**Comment by TanninOne** [comment]
No, absolutely not. The game directory is for game files, storing mods there is pretty horrible from a directory layout and security standpoint.
Storing mods in appdata or documents is the objectively right way to go.

Also, changing this now would require every user to go through a migration, I wouldn't do that for no practical purpose even _if_ I agreed that it would be cleaner.

Finally: You _can_ edit extensions, advanced mode or not. Edit the file, place the modified version in %APPDATA%\vortex\plugins\game-dragonage2, (re)start Vortex.

**Comment by ZeroMaki** [comment]
>game directory is for game files, mods there is horrible
>skyrim, mhw, any other game needs mods in game files

Well I suppose that makes perfect and absolute sense.. lets be real now here in appdata or documents is definitely not in any way objectively the way to go, because you're gonna have to query system drive even if symlink with the staging folder on the same drive as game, half the reason to not game on system drive is first of the drive is not used for anything but system and some programs like browser, but all game loading is on your gaming drive, documents and appdata gets bloated enough as is overtime, I don't see why security or other standpoints makes it in any way remotely better other than stuff can be hard to find and wipe versus in game folder and all you do is delete the game and its folder.
